lib/LoginHookListener.php

```diff
@@ -45,29 +45,62 @@ public function __construct(IConfig $config,
 		$this->isLoginPage = $isLoginPage;
 	}
 
-	/**
+        private function isIpv4($ip) {
+            return filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4);
+        }
+        
+        private function isIpv6($ip) {
+            return filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6);
+        }
+
+         /**
 	 * @param $ip
 	 * @param $range
 	 * @return bool
-	 * @copyright https://stackoverflow.com/questions/594112/matching-an-ip-to-a-cidr-mask-in-php-5/594134#594134
+	 * @copyright (IPv4) https://stackoverflow.com/questions/594112/matching-an-ip-to-a-cidr-mask-in-php-5/594134#594134
+	 * @copyright (IPv6) MW. https://stackoverflow.com/questions/7951061/matching-ipv6-address-to-a-cidr-subnet via https://www.phpclasses.org/browse/file/70429.html
 	 */
-	private function matchCidr($ip, $range) {
-		if ($range === '') {
```

Comment:
why is this gone?

Reply:
Not sure, it's been over a year. Probably should still be there.